libs/sdk-core/src/binding.rs

```diff
@@ -84,22 +93,29 @@ pub fn node_info() -> Result<NodeState> {
 /// Cleanup node resources and stop the signer.
 pub fn disconnect() -> Result<()> {
     block_on(async {
-        let shutdown_handler = BREEZ_SERVICES_SHUTDOWN.get();
-        match shutdown_handler {
-            None => Err(anyhow!("Background processing is not running")),
-            Some(s) => s.send(()).await.map_err(anyhow::Error::msg),
-        }
+        let mut locked_sdk_instance = BREEZ_SERVICES_INSTANCE.lock().await;
+        *locked_sdk_instance = None;
+
+        let mut locked_notifications_stream = NOTIFICATION_STREAM.lock().await;
+        *locked_notifications_stream = None;
+
+        let mut locked_log_init = LOG_INIT.lock().await;
+        *locked_log_init = None;
+
+        get_breez_services().await?.disconnect().await?;
```

Comment:
Does this work? I am pretty sure there is a deadlock here as the get_breez_services() aquiers the lock again...
Also why do we need to involve the NOTIFICATION_STREAM and the LOG_INIT ? can't these global handlers stay with OneCell?

Reply:
Good point. I shifted the `disconnect` above the locking, so they don't interfere.

> Also why do we need to involve the NOTIFICATION_STREAM and the LOG_INIT 

True. i thought they are set again after `disconnect`, but they cannot because `init_logging` sets the global logger, so it cannot be called again. I'll revert them.

Reply:
Done in 54b3b4810c5ffc04ef152f949b36c31f30b6670f